addons/lunch/models/lunch_alert.py

```diff
@@ -84,36 +87,92 @@ def _search_available_today(self, operator, value):
-            notification_from = notification_to - timedelta(minutes=5)
-            send_at = datetime.combine(fields.Date.today(),
-                float_to_time(alert.notification_time, alert.notification_moment))
-
-            if alert.available_today and send_at > notification_from and send_at <= notification_to:
-                order_domain = [('state', '!=', 'cancelled')]
 
-                if alert.location_ids.ids:
-                    order_domain = expression.AND([order_domain, [('user_id.last_lunch_location_id', 'in', alert.location_ids.ids)]])
-
-                if alert.recipients != 'everyone':
-                    weeks = 1
+    @api.depends('name', 'mode', 'notification_time', 'notification_moment', 'tz', 'active')
+    def _compute_cron_id(self):
+        for alarm in self:
+            alarm = alarm.with_context(tz=alarm.tz)
+
+            cron_required = (
+                alarm.active
+                and alarm.mode == 'chat'
+                and (not alarm.until or fields.Date.context_today(alarm) <= alarm.until)
+            )
+
+            now_tz = fields.Datetime.context_timestamp(alarm, datetime.utcnow())
+            sendat_tz = datetime.combine(date.today(), float_to_time(alarm.notification_time, alarm.notification_moment, alarm.tz))
+            if sendat_tz < now_tz:  # notif sent today already
+                sendat_tz += timedelta(days=1)
+            sendat_utc = sendat_tz.astimezone(pytz.UTC).replace(tzinfo=None)
+
+            alarm.cron_id.name = f"Lunch: alarm chat notification ({alarm.name})"
+            alarm.cron_id.active = cron_required
+            alarm.cron_id.nextcall = sendat_utc
+            alarm.cron_id.code = dedent(f"""\
+                # This cron is dynamically controlled by {self._description}.
+                # Do NOT modify this cron, modify the related record instead.
+                env['{self._name}'].browse([{alarm.id}])._notify_chat()""")
+
+
+    @api.model
+    def create(self, vals_list):
+        if isinstance(vals_list, dict):
+            vals_list = [vals_list]
+
+        crons = self.env['ir.cron'].sudo().create([
+            {
+                'user_id': self.env.ref('base.user_root').id,
+                'active': False,
+                'interval_type': 'days',
+                'interval_number': 1,
+                'numbercall': -1,
+                'doall': False,
+                'name': "Lunch: alarm chat notification",
+                'model_id': self.env['ir.model']._get_id(self._name),
+                'state': 'code',
+                'code': "",
+            }
+            for _ in range(len(vals_list))
+        ])
+        for vals, cron in zip(vals_list, crons):
+            vals['cron_id'] = cron.id
 
-                    if alert.recipients == 'last_month':
-                        weeks = 4
-                    else:  # last_year
-                        weeks = 52
+        alarms = super().create(vals_list)
+        alarms._compute_cron_id()
```

Comment:
Never call a compute method, as there is some necessary boilerplate for calling compute methods.
Simply mark the field to be computed on the given records:
```suggestion
        self.env.add_to_compute(self._fields['cron_id'], alarms)
```